content/articles/exception-handeling-in-windows-forms/index.md

```diff
@@ -0,0 +1,130 @@
+# Exception-handling-in-windows-forms
+
+### Introduction
+Windows forms are one of the best tools used in making applications, it provides a drag & drop functionality for a variety of pre-built objects such as buttons, textbox, drop-down selections, and many other useful tools, but how do we manage the inputs from the user?
+
+In this tutorial, we will manage a variety of errors & issues that would occur when using windows forms. We will use exception handling to prevent these errors and guide the user to use the application properly.
+
+### Prerequisites
+- A basic understanding of the C# programming language.
+- A basic understanding of Windows forms.
+- Visual Studio installed on your system.
+
+### Installation guide
+- To install Visual Studio and set up the work environment, you can check [this](https://www.geeksforgeeks.org/setting-environment-c-sharp/) article on how to set up the C# environment in Visual Studio.
+- If you are new to Windows forms, you can check [this](https://www.section.io/engineering-education/getting-started-with-windows-forms-using-c-sharp/) tutorial, which will help you understand the basic concepts.
+
+### Exception handling 
+Sometimes the program during runtime faces some errors that would ruin the process or the flow of the program, and here comes the concept of exception handling. With exception handling, we can test and fix all kinds of errors and return the flow to the program.
+
+We have three main keywords that we will be using when handling a piece of code in exception handling.
+- **Try:** scope is used for the piece of code that might possibly produce an error. It could also be used to throw an error intentionally to test out the code and see how it reacts to the error.
+- **Catch:** scope is used to catch the error and explain the issue to the user, and we use it to show a message box or request a data refill for the form. It could also be used to walk the user through the steps that would repair the program or maintain the issue, and sometimes we use it to change the value of an object or a variable.

```

Comment:
```suggestion
- **Catch:** scope is used to catch the error and explain the issue to the user, and we use it to show a message box or request a data refill for the form. It could also be used to walk the user through the steps that would repair the program or maintain the issue. Sometimes, we use it to change the value of an object or a variable.
```